Batch File/Batch File.sublime-syntax

```diff
@@ -9,10 +9,14 @@ file_extensions:
   - bat
   - cmd
 
+hidden_file_extensions:
+  - .bat
+  - .cmd
+
 first_line_match: |-
   (?xi:
-    ^ \s* (?:\:|rem) .*? -\*- .*? \bbat\b .*? -\*-  # editorconfig
-  | ^ \s* @echo\s+(?:on|off)\b
+    ^ [@\s]* (?:\:|rem) .*? -\*- .*? \b(?:bat|cmd)\b .*? -\*-  # editorconfig
+  | ^ [@\s]* (?:if \s+ \S .*? =?=? .*? \S \s+)? [@(\s]* echo \s+ (?:on|off|=) \b
```

Comment:
Not only `if a == b` but also `if 1 NEQ 0` are valid conditional expressions.

Your current pattern does match both, but that's not obvious. The `if ...` part may also suffer from catastrophic backtracking.

I'd therefore suggest the following simplification.

```suggestion
  | ^ [@\s]* (?:if \s .*? [@(\s] )? echo \s+ (?:on|off|=) \b
```

If the line starts with an `if` keyword, consume anything until `@echo`, `(echo`  or ` echo` are found. This should allow any kind of conditional expression.

Without `if` a line must start with `@echo` or ` echo` or ` @ @ @ echo`.

Reply:
Sounds good (and then `.*?` takes care of complex expressions anyway).

Reply:
Done.